merkle/tree_test.go

```diff
@@ -159,3 +157,32 @@ func Test_getSplitPoint(t *testing.T) {
 		require.EqualValues(t, tt.want, got, "getSplitPoint(%d) = %v, want %v", tt.length, got, tt.want)
 	}
 }
+
+func randBytes(size int) []byte {
+	b := make([]byte, size)
+	_, _ = crand.Read(b)
+	return b
+}
+
+// Contract: !bytes.Equal(input, output) && len(input) >= len(output)
+func mutateByteSlice(bytez []byte) []byte {
+	// If bytez is empty, panic
+	if len(bytez) == 0 {
+		panic("Cannot mutate an empty bytez")
+	}
+
+	// Copy bytez
+	mBytez := make([]byte, len(bytez))
+	copy(mBytez, bytez)
+	bytez = mBytez
+
+	// Try a random mutation
+	switch rand.Int() % 2 {
+	case 0: // Mutate a single byte
+		bytez[rand.Int()%len(bytez)] += byte(rand.Int()%255 + 1)
+	case 1: // Remove an arbitrary byte
+		pos := rand.Int() % len(bytez)
+		bytez = append(bytez[:pos], bytez[pos+1:]...)
+	}
+	return bytez
+}
```

Comment:
note for reviewers, copied from https://github.com/celestiaorg/celestia-core/blob/34b83b51889c92a9818221bb01dc78a35dc55c77/libs/test/mutate.go#L7-L28